controllers/k8ssandra/reaper_test.go

```diff
@@ -254,7 +254,10 @@ func createMultiDcClusterWithControlPlaneReaper(t *testing.T, ctx context.Contex
 			Name:      "reaper"},
 	}
 	t.Log("check that control plane reaper is created")
-	require.Eventually(f.ReaperExists(ctx, cpReaperKey), timeout, interval)
+	withReaper := f.NewWithReaper(ctx, cpReaperKey)
```

Comment:
Issue: Can we have an assertion here to ensure that the right ownerReferences are present? Can we also ensure that all the other resources you're modifying with the ownerReference have a test somewhere in the envtests to ensure that's getting placed?

Reply:
This one is actually a bit of a special case, because the control plane Reaper is not owned by a DC.

All other cases go through `f.ReaperExists()`, which now has the ownership check baked in.